core/src/main/java/org/infinispan/persistence/manager/TracedPersistenceManager.java

```diff
@@ -0,0 +1,117 @@
+package org.infinispan.persistence.manager;
+
+import java.util.concurrent.CompletionStage;
+import java.util.function.Predicate;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commands.write.WriteCommand;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.StoreConfiguration;
+import org.infinispan.container.entries.MVCCEntry;
+import org.infinispan.context.impl.TxInvocationContext;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.impl.ComponentRef;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.persistence.spi.MarshallableEntry;
+import org.infinispan.persistence.support.DelegatingPersistenceManager;
+import org.infinispan.telemetry.InfinispanSpan;
+import org.infinispan.telemetry.InfinispanSpanAttributes;
+import org.infinispan.telemetry.InfinispanTelemetry;
+import org.infinispan.telemetry.SpanCategory;
+import org.infinispan.transaction.impl.AbstractCacheTransaction;
+import org.infinispan.util.function.TriPredicate;
+
+@Scope(Scopes.NAMED_CACHE)
+public class TracedPersistenceManager extends DelegatingPersistenceManager {
+
+   public TracedPersistenceManager(PersistenceManager persistenceManager) {
+      super(persistenceManager);
+   }
+
+   @Inject
+   InfinispanTelemetry telemetryService;
+
+   @Inject
+   ComponentRef<AdvancedCache<Object, Object>> cache;
+
+   @Inject
+   Configuration configuration;
+
+   @Override
+   public CompletionStage<Void> purgeExpired() {
+      return decorateCommand("purgeExpired", super.purgeExpired());
+   }
+
+   @Override
+   public CompletionStage<Void> clearAllStores(Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("clearAllStores", super.clearAllStores(predicate));
+   }
+
+   @Override
+   public CompletionStage<Boolean> deleteFromAllStores(Object key, int segment,
+                                                       Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("deleteFromAllStores", super.deleteFromAllStores(key, segment, predicate));
+   }
+
+   @Override
+   public <K, V> CompletionStage<MarshallableEntry<K, V>> loadFromAllStores(Object key, int segment, boolean localInvocation, boolean includeStores) {
```

Comment:
we are not tracing reads in hot rod, rest or rpcs. Not sure why we are tracing them here.

Reply:
Right usually we don't trace read/load/query operations, but I thought that in case of persistence it was nice to trace it, since in order to optimize the persistence we usually want to minimize those operations.